app/models/manageiq/providers/redhat/infra_manager/event_parsing/parser.rb

```diff
@@ -40,6 +40,8 @@ def self.event_to_hash(event_obj, ems_id = nil)
         username = ems.ovirt_services.username_by_href(user_href)
       end
 
+      vm_ref = event[:name].include?("TEMPLATE") ? ems_ref_from_object_in_event(event[:template]) : ems_ref_from_object_in_event(event[:vm])
```

Comment:
> event[:name].include?("TEMPLATE")
Is there any way to do this differently? In any case I think this should be moved to a method and maybe when we find a better way to do it refactor it.